FilterList/FilterList.tsx

```diff
@@ -416,32 +417,51 @@ class FilterList extends Widget {
     });
   }
 
-  async setInitExpression(id: string, expression: Expression): Promise<void> {
+  setInitExpression(id: string, expression: Expression): void {
```

Comment:
This method is pretty massive, would it be possible to break this up into small pieces?